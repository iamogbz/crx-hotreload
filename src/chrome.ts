export const getExtensionInfo = () =>
    new Promise(resolve => chrome.management.getSelf(resolve));

export const getExtensionDirectory = () =>
    new Promise(resolve => chrome.runtime.getPackageDirectoryEntry(resolve));

export const getEntries = (dir: DirectoryEntry) =>
    new Promise(resolve => dir.createReader().readEntries(resolve));

export const reloadCurrentTab = () =>
    new Promise(resolve =>
        chrome.tabs.query({ active: true, currentWindow: true }, resolve),
    ).then(([tab]) => {
        if (tab) chrome.tabs.reload(tab.id);
        chrome.runtime.reload();
    });
