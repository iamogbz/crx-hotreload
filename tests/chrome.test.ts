import * as chrome from "sinon-chrome";

import { log } from "chrome";
import { SinonStub } from "sinon";

jest.spyOn(console, "log");
expect.extend({
    stubToHaveBeenCalledWith: (stub: SinonStub, ...args: any[]) => {
        const pass = stub.withArgs(...args).calledOnce;
        const prettyArgs = JSON.stringify(args).slice(1, -1);
        const message = `${stub} ${
            pass ? "" : "not "
        }called with (${prettyArgs})`;
        return { pass, message };
    },
});

describe("chrome", () => {
    afterEach(chrome.flush);
    describe("log", () => {
        const mockArgs = [1, "2", { three: "III" }, [4, "four", "IV", 4.0]];

        it("logs to console in background", () => {
            log(...mockArgs);
            // tslint:disable-next-line: no-console
            expect(console.log).toHaveBeenCalledWith(...mockArgs);
        });

        it("logs to console in activeTab", () => {
            log(...mockArgs);
            expect(chrome.tabs.executeScript.firstCall.args).toMatchSnapshot();
        });
    });
});
