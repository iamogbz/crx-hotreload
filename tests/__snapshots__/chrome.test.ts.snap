// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chrome log logs to console in activeTab 1`] = `
Array [
  Object {
    "code": "console.log(\`1\`,\`\\"2\\"\`,\`{\\"three\\":\\"III\\"}\`,\`[4,\\"four\\",\\"IV\\",4]\`)",
  },
]
`;
